apps/openmw/mwworld/esmstore.cpp

```diff
@@ -301,12 +302,14 @@ void ESMStore::setUp(bool validateRecords)
     if (validateRecords)
     {
         validate();
-        countRecords();
+        countAllCellrefs();
     }
 }
 
-void ESMStore::countRecords()
+void ESMStore::countAllCellrefs()
 {
+    // TODO: We currently need to read entire files here again.
+    // We should consider consolidating or deferring this reading.
```

Comment:
This might not be a useful comment. The problem is that to know how much times each Ref is present in the game, need to count presence of it in each cell. There is no separate index in ESM format saying where this Ref is present. When we first time call `getRefCount` for some id we need somehow to get the correct number. Caching this once on the game start is a good choice. Is there a benefit of deferring it? Could you think of possible solutions that will not cause reading ESM file in `getRefCount` and will give correct result?

Reply:
> Is there a benefit of deferring it? 

We do not need `getRefCount` until scripting system is accessed. We could do this reading in another thread as done in `ObjectPaging` for example. Alternatively, we could run a single reading pass that handles all of our requirements including `handleMovedCellRefs`. I think it is definitely a helpful comment because a `read` in an implementation of `count` is surely unexpected and worth noting. There is not much to add to the comment because possible approaches depend on other code that may change.